libr/asm/arch/whitespace/wsdis.c

```diff
@@ -310,22 +293,22 @@ WS_API int wsdis(RAsmOp *op, const ut8 *buf, int len) {
 			}
 			switch (*ptr) {
 			case 10:
-				sprintf (op->buf_asm, "ret");
+				r_strbuf_set (&op->buf_asm, "ret");
 				return op->size = ptr - buf + 1;
 			case '\t':
-				sprintf (op->buf_asm, "jn");
+				r_strbuf_set (&op->buf_asm, "jn");
 				break;
 			case ' ':
-				sprintf (op->buf_asm, "jz");
+				r_strbuf_set (&op->buf_asm, "jz");
 				break;
 			}
 			ptr++;
 			if (-1 == test_ws_token_exist (ptr, 10, len - (ptr - buf))) {
-				op->buf_asm[0] = '\0';
+				r_strbuf_set (&op->buf_asm, "");
```

Comment:
I'm puzzled why it is sometimes `"invalid"` sometimes empty

Reply:
when empty it's asm.c who sets the string "invalid". the idea is that when invalid, the backend should delegate this task to the frontend (r_asm), and we have 1 point to change if we want to change "invalid" for something else. same goes for unalisgned and such